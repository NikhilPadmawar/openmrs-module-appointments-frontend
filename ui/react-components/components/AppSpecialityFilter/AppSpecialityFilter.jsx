import React, { useState } from "react";
import { injectIntl } from "react-intl";
import PropTypes from "prop-types";
import classNames from "classnames";
import "./AppSpecialityFilter.module.scss";
import "rc-tree/assets/index.css";
import Tree from "rc-tree";
import { useEffect } from "react";

const rcTree = (props) => {
  const { nodes, getChecked, intl } = props;
  const [checked, setChecked] = useState([]);

  const switcherIcon = (obj) => {
    if (obj.isLeaf) return "";
    if (obj.expanded)
      return <i className={classNames("fa fa-minus")} aria-hidden="true"></i>;
    return <i className={classNames("fa fa-plus")} aria-hidden="true"></i>;
  };

  return (
    <div data-testid="tree">
      <Tree
        className={classNames("myCls")}
        checkable
        checkedKeys={checked}
        onCheck={(checked) => {
          getChecked(checked);
          setChecked(checked);
        }}
        treeData={nodes || []}
        switcherIcon={switcherIcon}
      />
    </div>
  );
};

rcTree.propTypes = {
  intl: PropTypes.object.isRequired,
};

export default injectIntl(rcTree);
