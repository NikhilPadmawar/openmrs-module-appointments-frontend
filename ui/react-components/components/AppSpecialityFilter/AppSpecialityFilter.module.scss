@import url("https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css");

.rc-tree .rc-tree-treenode span.rc-tree-switcher {
  margin-top: 5px;
  &.rc-tree-switcher_close {
    background: white;
  }
  &.rc-tree-switcher_open {
    background: white;
  }
}

.rc-tree-icon__customize {
  margin-bottom: 3px;
}
