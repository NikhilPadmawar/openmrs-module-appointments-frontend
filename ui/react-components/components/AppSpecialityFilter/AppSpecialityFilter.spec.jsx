import React from "react";
import { render, wait } from "@testing-library/react";
import AppSpecialityFilter from "./AppSpecialityFilter";
import { renderWithReactIntl } from "../../utils/TestUtil";
import "@testing-library/jest-dom/extend-expect";

describe("Should check length", () => {
  it("Check if nodeList is empty", async () => {
    const nodes = [];

    const { container } = await renderWithReactIntl(
      <AppSpecialityFilter nodes={nodes} />
    );
    expect(container.getElementsByClassName("rc-tree-treenode").length).toBe(
      nodes.length
    );
  });
});

describe("Tree should be visible", () => {
  it("should be visible", async () => {
    const { getByTestId } = await renderWithReactIntl(<AppSpecialityFilter />);
    expect(getByTestId("tree")).toBeInTheDocument();
  });
});

describe("Checked if class is present", () => {
  it("checkable", async () => {
    const { asFragment } = await renderWithReactIntl(
      <AppSpecialityFilter checkable></AppSpecialityFilter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
