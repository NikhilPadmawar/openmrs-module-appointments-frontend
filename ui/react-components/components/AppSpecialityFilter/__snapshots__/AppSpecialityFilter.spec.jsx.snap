// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checked if class is present checkable 1`] = `
<DocumentFragment>
  <div
    data-testid="tree"
  >
    <div
      class="rc-tree myCls"
    >
      <div
        role="tree"
      >
        <input
          style="width: 0px; height: 0px; display: flex; overflow: hidden; opacity: 0; border: 0px; padding: 0px; margin: 0px;"
          tabindex="0"
          value=""
        />
      </div>
      <div
        class="rc-tree-list"
      >
        <div>
          <div
            class="rc-tree-list-holder-inner"
            style="display: flex; flex-direction: column;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
